src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,132 @@
+                    + manufacturer, e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteManufacturerRequest =
+                "UPDATE manufacturers SET is_deleted = true WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteManufacturerStatement =
+                        connection.prepareStatement(deleteManufacturerRequest)) {
+            deleteManufacturerStatement.setLong(1, id);
+            return deleteManufacturerStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t delete manufacturer from DB by id: " + id, e);
+        }
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        String updateManufacturerRequest =
+                "UPDATE manufacturers SET name = ?, country = ? "
+                        + "WHERE is_deleted = false AND id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateManufacturerStatement =
+                        connection.prepareStatement(updateManufacturerRequest)) {
+            updateManufacturerStatement.setString(1, manufacturer.getName());
+            updateManufacturerStatement.setString(2, manufacturer.getCountry());
+            updateManufacturerStatement.setLong(3, manufacturer.getId());
+            Optional<Manufacturer> oldManufacturer = get(manufacturer.getId());
+            updateManufacturerStatement.executeUpdate();
+            if (oldManufacturer.isPresent()) {
+                return oldManufacturer.get();
+            }
+            throw new RuntimeException("Can`t find manufacturer in DB by id: "
+                    + manufacturer.getId());
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t update manufacturer: "
+                    + manufacturer, e);
+        }
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        String getManufacturerRequest =
+                "SELECT * FROM manufacturers WHERE is_deleted = false AND id = ?;";
+        Manufacturer manufacturer = new Manufacturer();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getManufacturerStatement =
+                        connection.prepareStatement(getManufacturerRequest)) {
+            getManufacturerStatement.setLong(1, id);
+            ResultSet resultSet = getManufacturerStatement.executeQuery();
+            checkResultSet(resultSet);
+            while (resultSet.next()) {
+                manufacturer = initializeManufacturer(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get manufacturer from DB by id: " + id, e);
+        }
+        return Optional.of(manufacturer);
```

Comment:
```suggestion
        Manufacturer manufacturer = null;
        try (Connection connection = ConnectionUtil.getConnection();
                PreparedStatement getManufacturerStatement =
                        connection.prepareStatement(getManufacturerRequest)) {
            getManufacturerStatement.setLong(1, id);
            ResultSet resultSet = getManufacturerStatement.executeQuery();
            checkResultSet(resultSet);
            while (resultSet.next()) {
                manufacturer = initializeManufacturer(resultSet);
            }
        } catch (SQLException e) {
            throw new DataProcessingException("Can`t get manufacturer from DB by id: " + id, e);
        }
        return Optional.ofNullable(manufacturer);
```

as we should get empty Optional in case there is no manufacturer with such id